Dockerfile

```diff
@@ -42,8 +21,7 @@ ENV GOCACHE=/go/cache
 
 # Download go modules
 RUN --mount=type=cache,target=/go go mod download
-# Do not verify modules if we have local modules coming from go.work
-RUN --mount=type=cache,target=/go if [ "${WITH_GO_WORK}" = "0" ]; then go mod verify; fi
+RUN --mount=type=cache,target=/go go mod verify
```

Comment:
I would remove the cache mount inside the production Dockerfile altogether, just run plain go commands.
That way we are sure there are no poisoned caches around.